code/modules/vehicles/cargo_train.dm

```diff
@@ -360,9 +360,9 @@
 // more engines increases this limit by car_limit per
 // engine.
 //-------------------------------------------------------
-/obj/vehicle/train/cargo/engine/update_car(var/train_length, var/active_engines)
-	src.train_length = train_length
-	src.active_engines = active_engines
+/obj/vehicle/train/cargo/engine/update_car(train_length, active_engines)
```

Comment:
I would keep these src. because parameters are same or change parameter names for something sane